core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -1304,7 +1304,31 @@ public final Sql sql(String sql) {
     final String sql = "select d.deptno, e2.empno\n"
         + "from dept_nested as d,\n"
         + " UNNEST(d.employees) as e2(empno, y, z)";
-    sql(sql).with(getExtendedTester()).ok();
+    sql(sql).ok();
+  }
+
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3789">[CALCITE-3789]
+   * Support validation of UNNEST multiple array columns like Presto</a>.
+   */
+  @Test public void testAliasUnnestArrayPlanWithSingleColumn() {
+    final String sql = "select d.deptno, employee.empno\n"
+        + "from dept_nested_expanded as d,\n"
+        + " UNNEST(d.employees) as t(employee)";
+    sql(sql).conformance(SqlConformanceEnum.PRESTO).decorrelate(false).ok();
```

Comment:
The reason of turning off `decorrelate` is because during flattening, the row type of Uncollect is not fully flattened, but the index is calculated on a fully flattened base (RelStructuredTypeFlattener.postFlattenSize). So the type mismatches validated query. Not sure whether it will be a big issue tho.

Reply:
It is, you should fix the `RelStructuredTypeFlattener`.

Reply:
Done in @danny0405 patch. Will resolve this comment.